modyn/selector/internal/selector_strategies/downsampling_strategies/abstract_downsampling_strategy.py

```diff
@@ -42,7 +42,6 @@ def __init__(
         self.requires_remote_computation = True
         self.maximum_keys_in_memory = maximum_keys_in_memory
         self.downsampling_config = downsampling_config
-        self.downsampling_params = self._build_downsampling_params()
```

Comment:
In the case of `RHOLossDownsampler`, the `downsampling_params` contains `il_model_id`, which is dynamic (contingent on trigger id). So we cannot attach a static field.

Reply:
I wonder if we can cache the value using @cached_property by default. However, the problem is we'd need to invalidate this in the RHO-LOSS subclass, and I could not find out if that's possible. Probably calling the function on other classes is not expensive at all, so it should be fine

Reply:
Yeah I wouldn't worry too much about an additional function call per trigger. The returned data is a small dict which takes no overhead to compute (if the returned data is computed & fetched from disk, then definitely we need to cache it)